app/models/abstract_model.rb

```diff
@@ -150,6 +152,21 @@ def self.count_by_sql_wrapping_select_query(sql)
     count_by_sql("select count(*) from (#{sql}) as my_table")
   end
 
+  # Lookup all instances matching a given wildcard pattern.  If there are no
+  # "*" in the pattern, it just does a regular find_by_xxx lookup.  A number
+  # of convenience wrappers are included in the major models. Returns nil if
+  # none found.
+  #
+  #   Project.find_using_wildcards("title", "FunDiS *")
+  #   Project.find_by_title_with_wildcards("FunDiS *")
+  #
+  def self.find_using_wildcards(col, str)
+    return send("find_by_#{col}", str) unless str.include?("*")
+
+    matches = where("#{col} LIKE ?", str.tr("*", "%"))
```

Comment:
How does one fix this?  "col" is supplied by our code, so it is guaranteed to be safe.  But it may be easier to do something like:

    where("? LIKE ?", col, str)

Is that allowed?  Why not, I suppose.  I'll give it a shot.

Reply:
Hmm, don't think that will work since `#{col}` is a column name.  I think `?` only works for values.

Reply:
Brakeman is pointing out something that could potential be an issue.  I'd resolve it with a lookup table since you are currently only calling it explicitly with "code", "name" or "title".  So something like:

```
   matches = where(CLAUSES_FOR_COL[col], str.tr("*", "%"))
```

where

`CLAUSES_FOR_COL = ["code", "name", "title"].each_with_object({}) { |col, memo| memo[col] = "#{col} LIKE ?" }`

should work.  I'm not sure if brakeman will think that's cool, but the point is to make sure that `col` is only from an expected set of values that you actual know ahead of time.  My proposed fix will raise an error if someone tries it with values that aren't expected.  Then can then decide to either extend the set of possible values or take another approach.

Reply:
I have no idea where my comments on this thread went.  I have fixed in a different way.  There is a method on Model.connection that escapes a value for use as a column name.  I use this to escape the column name in the line above.  Not sure if brakeman will accept it, but it is 100% foolproof.  (Well, it was already 100% foolproof!  But now the code ensuring it is *right there*, no need to check that all agents calling this method have untainted the column name.)